source/opt/def_use_manager.cpp

```diff
@@ -38,10 +38,14 @@ namespace analysis {
 
 void DefUseManager::AnalyzeInstDefUse(ir::Instruction* inst) {
   const uint32_t def_id = inst->result_id();
-  // Clear the records of def_id first if it has been recorded before.
-  ClearDef(def_id);
+  if (def_id != 0) {
+    ClearDef(def_id);
+    id_to_def_[def_id] = inst;
+  } else {
+    ClearInst(inst);
+  }
 
-  if (def_id != 0) id_to_def_[def_id] = inst;
+  inst_to_used_ids_[inst].reserve(inst->NumOperands());
```

Comment:
I don't think we need this. This will highly likely reserve more than what's needed.

Most of the instructions take <= 2 operand IDs and I think `std::vector` should already allocate enough space and optimize that case.